src/libertem/udf/pca.py

```diff
@@ -0,0 +1,217 @@
+                ),
+            'mean': self.buffer(
+                kind='sig', dtype='float32'
+                ),
+            'var': self.buffer(
+                kind='sig', dtype='float32'
+                ),
+            'num_frame': self.buffer(
+                kind='single', dtype='float32'
+                ),
+            'singular_vals': self.buffer(
+                kind='sig', dtype='float32'
+                )
+            }
+
+    def merge(self, dest, src):
+        """
+        Given two sets of partitions, with number of components,
+        mean, variance, number of frames used, singular values, and
+        explained variance (by singular values), update the joint
+        mean, variance, number of frames used, singular values, and
+        explained variance
+
+        Parameters
+        ----------
+        dest
+            Contains information about the first partition, including
+            sum of variances, sum of pixels, and number of frames used
+
+        src
+            Contains information about the second partition, including
+            sum of variances, sum of pixels, and number of frames used
+
+        pca
+            colletions.namedtuple object that contains information about
+            the merged partitions, including mean, variances,
+            number of frames used, singular values for partitions, and
+            the number of components
+        """
+        prev = pca(
+                    var=dest['var'][:],
+                    mean=dest['mean'][:],
+                    num_frame=dest['num_frame'][:],
+                    singular_val=dest['singular_vals'][:],
+                    n_components=dest['n_compoennts'][:]
+                    )
+        new = pca(
+                    var=src['var'][:],
+                    mean=src['mean'][:],
+                    num_frame=src['num_frame'][:],
+                    singular_val=src['singular_vals'][:],
+                    n_components=src['n_compoennts'][:]
+                    )
+
+        compute_merge = IncrementalPCA(prev, new)
+
+        dest['var'][:] = compute_merge.var
+        dest['mean'][:] = compute_merge.sum_im / compute_merge.N
+        dest['num_frame'][:] = compute_merge.N
+        dest['singular_vals'][:] = compute_merge.singular_vals
+        dest['n_components'][:] = compute_merge.n_components
+
+    def process_tile(self, tile):
+        """
+        Given a tile, update parameters related to PCA
+
+        Parameters
+        ----------
+        tile
+            single tile of the data
+        """
+        prev = pca(
+            sum_im=self.results.mean * self.results.N,
+            var=self.results.var,
+            N=self.results.num_frame,
+            n_components=self.results.n_components,
+            singular_values=self.results.singular_vals
+            )
+
+        sig_dim = self.results.singular_vals.shape
+        num_frame = tile.shape[0]
+
+        new = pca(
+            sum_im=np.sum(tile),
```

Comment:
You may need to specify `axis=0` here, right? Otherwise, the whole tile is reduced to a scalar value

Reply:
Yes totally